src/components/Tabs/Scroller.tsx

```diff
@@ -0,0 +1,236 @@
+import { css } from 'glamor'
+import React, { useRef, useState, useEffect } from 'react'
+import scrollIntoView from 'scroll-into-view'
+import { ChevronLeftIcon, ChevronRightIcon } from '../Icons'
+import { plainButtonRule } from '../Button'
+import { useColorContext } from '../Colors/useColorContext'
+import { mUp } from '../../theme/mediaQueries'
+
+const styles = {
+  container: css({
+    position: 'relative',
+    overflow: 'hidden'
+  }),
+  breakoutMargin: css({
+    [mUp]: {
+      margin: 0
+    }
+  }),
+  scroller: css({
+    display: 'flex',
+    flexDirection: 'row',
+    overflowX: 'scroll',
+    flexWrap: 'nowrap',
+    scrollbarWidth: 'none' /* Firefox */,
+    msOverflowStyle: 'none' /* IE 10+ */,
+    WebkitOverflowScrolling: 'touch',
+    '::-webkit-scrollbar': {
+      display: 'none'
+    }
+  }),
+  arrow: css(plainButtonRule, {
+    display: 'none',
+    '@media (hover)': {
+      display: 'flex',
+      position: 'absolute',
+      top: 0,
+      bottom: 0,
+      textAlign: 'center',
+      alignItems: 'center',
+      justifyContent: 'center',
+      pointerEvents: 'none',
+      opacity: 0,
+      transition: 'opacity 200ms'
+    }
+  }),
+  arrowIcon: css({
+    // ontop of arrowBg
+    position: 'relative'
+  }),
+  arrowBg: css({
+    display: 'block',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+    bottom: 0,
+    opacity: 0.7
+  }),
+  arrowHoverable: css({
+    '@media (hover)': {
+      '[role=group]:hover > &': {
+        pointerEvents: 'auto',
+        opacity: 1
+      }
+    }
+  })
+}
+
+type ScrollerType = {
+  children: React.ReactNode
+  center?: boolean
+  activeChildIndex?: number
+  hideArrows?: boolean
+  arrowSize?: number
+  breakoutPadding?: number
+}
+
+const Scroller = ({
+  children,
+  activeChildIndex = 0,
+  center = false,
+  hideArrows = false,
+  arrowSize = 28,
+  breakoutPadding = 0
+}: ScrollerType) => {
+  const scrollRef = useRef<HTMLDivElement>()
+  const [{ left, right }, setArrows] = useState({ left: false, right: false })
+  const [scrollerOverflow, setOverflow] = useState(false)
```

Comment:
This can just be deferred from the arrows afais:
```
  const hasOverflow = left || right
```

Reply:
that was the previous way it was done, which led you to assume that the overflowPadding was dependent on the visibility of the arrows XD. So what's it going to be? This is a bit more declarative and understandable imho :D